File: app/src/main/java/pl/gocards/ui/decks/xml/standard/dialog/NoCardsToRepeatDialog.java
package pl.gocards.ui.decks.xml.standard.dialog;

import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import com.google.android.material.dialog.MaterialAlertDialogBuilder;

import pl.gocards.R;

/**
 * D_R_04 No cards to repeat
 * @author Grzegorz Ziemski
 */
public class NoCardsToRepeatDialog extends DialogFragment {

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        return new MaterialAlertDialogBuilder(requireActivity())
                .setTitle(R.string.decks_list_no_more_cards_repeat)
                .setPositiveButton(R.string.ok, (dialog, which) -> {})
                .create();
    }

}

File: app/src/main/java/pl/gocards/ui/decks/decks/dialogs/NoCardsToRepeatDialog.kt
package pl.gocards.ui.decks.decks.dialogs

import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.res.stringResource
import pl.gocards.R

/**
 * D_R_04 No cards to repeat
 * @author Grzegorz Ziemski
 */
@Composable
fun NoCardsToRepeatDialog(
    onDismiss: () -> Unit
) {
    AlertDialog(
        title = { Text(text = stringResource(R.string.decks_list_no_more_cards_repeat)) },
        onDismissRequest = {
            onDismiss()
        },
        confirmButton = {
            TextButton(
                onClick = {
                    onDismiss()
                }
            ) {
                Text(stringResource(R.string.ok))
            }
        }
    )
}